pkg/options/common.go

```diff
@@ -12,37 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package maker
+package options
 
 import (
 	"fmt"
 
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	bundle "github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/apis/bundle/v1alpha1"
-	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/filter"
 )
 
-// ObjHandler is a function that can 'make' objects.
-type ObjHandler func(obj *unstructured.Unstructured, ref bundle.ComponentReference, pm ParamMaker) (*unstructured.Unstructured, error)
+// ObjHandler is a function that can apply options to a specific Kubernetes
+// object.
+type ObjHandler func(obj *unstructured.Unstructured, ref bundle.ComponentReference, pm OptionsFn) (*unstructured.Unstructured, error)
 
-// MakeCommon provides common functionality for making components, deferring
-// only the object handling logic.
-func MakeCommon(comp *bundle.ComponentPackage, p ParamMaker, of *filter.Options, fn ObjHandler) (*bundle.ComponentPackage, error) {
+// ApplyCommon provides common functionality for applying options, deferring
+// the specific object handling logic.
+func ApplyCommon(comp *bundle.ComponentPackage, optFn OptionsFn, objFn ObjHandler) (*bundle.ComponentPackage, error) {
```

Comment:
That was the intent. But the raw text package can't work that way because one ConfigMap can contain multiple objects. So, I think having this helper is maybe a bit more workable in the general case

Reply:
But you're not using `ApplyCommon` in `RawTextApplier`, afaict.

Reply:
Yeah, that's correct. I created the Common thing because i was thinking about the PatchTemplateApplier that I'm thinking about. Maybe this is over-design, but it's not too-far off I think.